include/pybind11/detail/common.h

```diff
@@ -151,9 +203,9 @@
 
 /// Include Python header, disable linking to pythonX_d.lib on Windows in debug mode
 #if defined(_MSC_VER)
-#    pragma warning(push)
+PYBIND11_WARNING_PUSH
+PYBIND11_DISABLE_WARNING_MSVC(4505)
```

Comment:
Sorry I missed this before, I have a slight preference for

```
PYBIND11_WARNING_DISABLE_MSVC
```

etc. so that we have a family of very obviously related macros:

```
PYBIND11_WARNING_PUSH
PYBIND11_WARNING_POP
PYBIND11_WARNING_DISABLE_...
```

But feel free to override my slight preference.

Reply:
Good point and good suggestion! Done.